File: Frontend/ready2road/src/app/model/Utente.ts
import {Wallet} from "./Wallet";
import {Inject} from "@angular/core";
import {TratteService} from "../services/tratte.service";
import {UtenteService} from "../services/utente.service";


export class Utente{
  private  nome: string;
  private  cognome: string;
  private  indirizzoEmail: string;
  private  numeroTelefono: string;
  private  dataNascita: Date;
  private  password: string;
  private ban: boolean;
  private  wallet: Wallet|undefined;
    getNome= (): string =>{
        return this.nome;

    }
    getCognome= (): string =>{
        return this.cognome;

    }
    getIndirizzoEmail= (): string =>{
        return this.indirizzoEmail;

    }
    getNumeroTelefono= (): string =>{
        return this.numeroTelefono;

    }
    getDataNascita= (): Date =>{
        return this.dataNascita;

    }
    getPassword= (): string =>{
        return this.password;

    }


    constructor(json:any, @Inject(UtenteService) private service: UtenteService) {
        this.nome = json.nome;
        this.cognome = json.cognome;
        this.indirizzoEmail = json.indirizzoEmail;
        this.numeroTelefono = json.numeroTelefono;
        this.dataNascita = json.dataNascita;
        this.password = json.password;
        this.ban = json.ban;
    }
    setNome= (nome: string): void =>{
        this.nome=nome;

    }

    setCognome= (cognome: string): void =>{
        this.cognome=cognome;

    }
    setIndirizzoEmail= (indirizzoEmail: string): void =>{
        this.indirizzoEmail=indirizzoEmail;

    }
    setNumeroTelefono= (numeroTelefono: string): void =>{
        this.numeroTelefono=numeroTelefono;

    }
    setDataNascita= (dataNascita: Date): void =>{
        this.dataNascita=dataNascita;

    }
    setPassword= (password: string): void =>{
        this.password=password;

    }
    setWallet= (wallet: Wallet): void =>{
        this.wallet=wallet;

    }
    getBan= (): boolean =>{
        return this.ban;

    }
    setBan= (ban: boolean): void =>{
        this.ban=ban;

    }

}

File: Ready2RoadBackend/src/main/java/Ready2road/Persistenza/Model/Utente.java
package Ready2road.Persistenza.Model;

import com.google.gson.JsonObject;

import java.util.Date;

public class Utente {

    private String indirizzoEmail, cognome, nome, password, numeroTelefono;
    private Boolean ban;

    public Boolean isBan() {
        return ban;
    }

    public void setBan(Boolean ban) {
        this.ban = ban;
    }

    private Date dataNascita;
    private Wallet wallet;

    public String getIndirizzoEmail() {
        return indirizzoEmail;
    }

    public void setIndirizzoEmail(String indirizzoEmail) {
        this.indirizzoEmail = indirizzoEmail;
    }

    public String getCognome() {
        return cognome;
    }

    public void setCognome(String cognome) {
        this.cognome = cognome;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getNumeroTelefono() {
        return numeroTelefono;
    }

    public void setNumeroTelefono(String numeroTelefono) {
        this.numeroTelefono = numeroTelefono;
    }

    public Date getDataNascita() {
        return dataNascita;
    }

    public void setDataNascita(Date dataNascita) {
        this.dataNascita = dataNascita;
    }

    public Wallet getWallet() {
        return wallet;
    }

    public void setWallet(Wallet wallet) {
        this.wallet = wallet;
    }


    public JsonObject toJson() {
        JsonObject jsonObject=new JsonObject();
        jsonObject.addProperty("nome",this.getNome());
        jsonObject.addProperty("cognome",this.getCognome());
        jsonObject.addProperty("numeroTelefono",this.getNumeroTelefono());
        jsonObject.addProperty("dataNascita",this.getDataNascita().toString());
        jsonObject.addProperty("password",this.getPassword());
        jsonObject.addProperty("indirizzoEmail",this.getIndirizzoEmail());
        jsonObject.addProperty("ban",this.isBan());
        jsonObject.add("wallet", this.getWallet().toJson());
        return jsonObject;
    }
}

